File: src/main/java/com/cocktailsguru/app/cocktail/dto/list/CocktailListItemDto.java
package com.cocktailsguru.app.cocktail.dto.list;

import com.cocktailsguru.app.comment.domain.Comment;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigInteger;
import java.util.List;

@Data
@NoArgsConstructor
public class CocktailListItemDto {
    private BigInteger id;
    private String name;
    private String image;
    private double calculatedRating;
    private int alcoVolume;
    private int numOfFavorite;
    private List<Comment> commentList;
}

File: src/main/java/com/cocktailsguru/app/cocktail/dto/list/CocktailListItemDto.kt
package com.cocktailsguru.app.cocktail.dto.list

import com.cocktailsguru.app.cocktail.domain.Cocktail

data class CocktailListItemDto(
        val id: Long,
        val name: String,
        val image: String,
        val calculatedRating: Double,
        val alcoVolume: Int,
        val numOfFavorite: Int
) {
    constructor(cocktail: Cocktail) : this(
            cocktail.id,
            cocktail.name,
            cocktail.imageName,
            cocktail.calculatedRating,
            cocktail.alcoVolume,
            cocktail.numOfFavorite
    )
}